components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -438,3 +450,106 @@ func InjectCertConfig(notebook *nbv1.Notebook, configMapName string) error {
 	}
 	return nil
 }
+
+// SetContainerImageFromRegistry checks if there is an internal registry and takes the corresponding actions to set the container.image value.
+// If an internal registry is detected, it uses the default values specified in the Notebook Custom Resource (CR).
+// Otherwise, it checks the last-image-selection annotation to find the image stream and fetches the image from status.dockerImageReference,
+// assigning it to the container.image value.
+func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, notebook *nbv1.Notebook, log logr.Logger) error {
+    // Create a dynamic client
+    dynamicClient, err := dynamic.NewForConfig(config)
+    if err != nil {
+        log.Error(err, "Error creating dynamic client")
+        return err
+    }
+    // Specify the GroupVersionResource for imagestreams
+    ims := schema.GroupVersionResource{
+        Group:    "image.openshift.io",
+        Version:  "v1",
+        Resource: "imagestreams",
+    }
+
+    annotations := notebook.GetAnnotations()
+    if annotations != nil {
+        if imageSelection, exists := annotations["notebooks.opendatahub.io/last-image-selection"]; exists {
+            // Check if the image selection has an internal registry, if so  will pickup this. This value constructed on the initialization of the Notebook CR.
+            if strings.Contains(notebook.Spec.Template.Spec.Containers[0].Image, "image-registry.openshift-image-registry.svc:5000") {
```

Comment:
Another question - are we sure that the value for the internal registry will always be this way? Is there a possibility that there could be some custom value?

Why don't we want to check presence of internal registry via ImageStream `status.dockerImageRepository` value as empty (no internal registry enabled) vs non-empty (internal registry present)? This seem to me as a more proper approach than to check with what the Notebook CR was created :thinking:

Reply:
agreed, that is what the image content change trigger admission plugin, which we are not using up to now (part of PR 800) is doing on the openshift side as well.